### PR TITLE
[BEAM-5366] Vendor gRPC and its transitive dependencies as a separate package.

### PR DESCRIPTION
**Pull request by lukecwik**
This prevents beam-model-pipeline, beam-model-fn-execution and beam-model-job...
from having copies of gRPC and their transitive dependencies being published
as artifacts within Apache Beam.

Note that this is known to break the Dataflow import process since we
are defining a new Gradle project and can only be fixed up during
the actual import.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by lukecwik** [comment]
CC: @charlesccychen I'll work on providing a change to the person doing the import that is associated with this as this can only be fixed up during import.

**Comment by charlesccychen** [comment]
Thanks @lukecwik.  CC: @aaltay regarding the import.

**Comment by aaltay** [comment]
Thank you both. Please let me know when it is ready to merge.

**Comment by lukecwik** [comment]
Busy with other responsibilities, unlikely to be able to get this for a little while to fix up Dataflow.

**Comment by lukecwik** [comment]
R: @swegner 
R: @boyuanzz

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 1]
```diff
@@ -1314,63 +1314,94 @@ artifactId=${project.name}
 
     /** ***********************************************************************************************/
 
+    // We specifically export the shadow closure and the dependencies closure so that they
+    // can be re-used by the beam-vendor-java-grpc-v1 project and within the applyPortabilityNature
+    // block.
+    project.ext.GRPC_V1_SHADOW_CLOSURE = {
```
Comment (kennknowles) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: If this is used in just one place ever, can it be inlined?

Comment (lukecwik) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: It is used in two places.

Comment (kennknowles) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: Ah, I see where & why. Thanks!

**Pull request merged**